packages/storage/src/indexed-db/index.ts

```diff
@@ -824,4 +817,43 @@ export class IndexedDb implements IndexedDbInterface {
       output: Value.fromJson(result.output),
     };
   }
+
+  fetchStakingTokenId(): AssetId {
+    const registryClient = new ChainRegistryClient();
+    const registry = registryClient.get(this.chainId);
+    const stakingToken = registry.stakingAssetId;
+
+    return stakingToken;
+  }
+
+  async hasStakingAssetBalance(assetId: AssetId): Promise<boolean> {
+    const spendableUMNotes = await this.db.getAllFromIndex(
+      'SPENDABLE_NOTES',
+      'assetId',
+      uint8ArrayToBase64(assetId.inner),
+    );
+
+    // Iterate over the spendable UM notes, and accrue balance for unspent notes
+    const stakingTokenBalance = new Amount();
+    for (const note of spendableUMNotes) {
+      const umNote = SpendableNoteRecord.fromJson(note);
+      if (umNote.heightSpent === 0n) {
+        const noteAmount = addLoHi(
+          { lo: stakingTokenBalance.lo, hi: stakingTokenBalance.hi },
+          {
+            lo: BigInt(umNote.note?.value?.amount?.lo ?? 0n),
+            hi: BigInt(umNote.note?.value?.amount?.hi ?? 0n),
+          },
+        );
+        stakingTokenBalance.lo = noteAmount.lo;
+        stakingTokenBalance.hi = noteAmount.hi;
+      }
+    }
+
+    if (stakingTokenBalance.lo > 0) {
+      return true;
+    }
+
+    return false;
+  }
```

Comment:
Can we simplify this?

```suggestion
async hasStakingAssetBalance(assetId: AssetId): Promise<boolean> {
  const spendableUMNotes = await this.db.getAllFromIndex(
    'SPENDABLE_NOTES',
    'assetId',
    uint8ArrayToBase64(assetId.inner),
  );

  return spendableUMNotes.some(note => {
    const umNote = SpendableNoteRecord.fromJson(note);
    return umNote.heightSpent === 0n && !isZero(getAmountFromRecord(umNote));
  });
}
```

Reply:
this is better, returning when first note balance is detected. changed